app/client/templates/elements/compileContract.js

```diff
@@ -59,10 +59,9 @@ Template['elements_compileContract'].onCreated(function() {
         var selectedToken = TemplateVar.getFrom('.select-token', 'selectedToken');
         var mainRecipient = TemplateVar.getFrom('div.dapp-address-input input.to', 'value');
         var replayProtectionOn = TemplateVar.get('replay-protection-checkbox');
+        var selectedType = TemplateVar.get('selectedType');
 
-        console.log('autorun', selectedContract, replayProtectionOn, selectedToken )
-
-        if(selectedContract){        
+        if(selectedType == 'source-code' && selectedContract){        
```

Comment:
this should be a tripple `===`, which means same content and type.
Its just good practice and would throw an error in ads lint